File: 20200807-python-doctest/fizzbuzzer.py
def fizz_buzz(number):
    """ FizzBuzzes the given number.
    >>> fizz_buzz(2)
    '2'
    >>> fizz_buzz(3)
    'Fizz'
    >>> fizz_buzz(4)
    '4'
    >>> fizz_buzz(5)
    'Buzz'
    >>> fizz_buzz(15)
    'FizzBuzz'
    """
    if multiple_of_3(number) and multiple_of_5(number):
        return "FizzBuzz"
    if multiple_of_3(number):
        return "Fizz"
    if multiple_of_5(number):
        return "Buzz"
    return str(number)


def multiple_of_3(number):
    return number % 3 == 0


def multiple_of_5(number):
    return number % 10 == 0

File: 20210325-kotlin-fizzbuzz/src/main/kotlin/com/katas/FizzBuzzer.kt
package com.katas

class FizzBuzzer {

    fun fizzBuzz(number: Int): String = when {
        isFizzBuzz(number) -> "FizzBuzz"
        isFizz(number) -> "Fizz"
        isBuzz(number) -> "Buzz"
        else -> number.toString()
    }

    private fun isFizzBuzz(number: Int) = isFizz(number) && isBuzz(number)

    private fun isFizz(number: Int) = number % 3 == 0

    private fun isBuzz(number: Int) = number % 5 == 0
}
